misc/snapcraft_releases.py

```diff
@@ -0,0 +1,109 @@
+    Beta = "beta"
+    Edge = "edge"
+
+
+CHANNELS = (Channel.Stable, Channel.Candidate, Channel.Beta, Channel.Edge)
+
+
+def get_tracks_for_version(version: Version) -> list[Track]:
+    TRACKS_FOR_V2 = (Track.V2,)
+    # TODO: Add Latest track once we publicly release v3
+    TRACKS_FOR_V3 = (Track.V3,)
+
+    tracks: list[Track] = []
+    match version.major:
+        case 2:
+            tracks.extend(TRACKS_FOR_V2)
+        case 3:
+            tracks.extend(TRACKS_FOR_V3)
+        case _:
+            raise ValueError(f"Unsupported major version: {version.major}")
+    return tracks
+
+
+assert get_tracks_for_version(Version.parse("2.0.0")) == [Track.V2]
+assert get_tracks_for_version(Version.parse("3.0.0")) == [Track.V3]
+
+
+def get_channel_for_version(version: Version) -> Channel:
+    PRERELEASE_TO_CHANNEL = {
+        "a": Channel.Edge,
+        "b": Channel.Beta,
+        "rc": Channel.Candidate,
+    }
+
+    if version.prerelease:
+        for pre, chan in PRERELEASE_TO_CHANNEL.items():
+            if version.prerelease.startswith(pre):
+                return chan
+        raise ValueError(f"Unsupported prerelease: {version.prerelease}")
+    elif version.dev is not None:
+        return Channel.Edge
+    else:
+        return Channel.Stable
+
+
+assert get_channel_for_version(Version.parse("3.0.0")) == Channel.Stable
+assert get_channel_for_version(Version.parse("3.0.0-a.0")) == Channel.Edge
+assert get_channel_for_version(Version.parse("3.0.0-b.0")) == Channel.Beta
+assert get_channel_for_version(Version.parse("3.0.0-rc.0")) == Channel.Candidate
+assert get_channel_for_version(Version.parse("3.0.0-dev.0")) == Channel.Edge
+
+if __name__ == "__main__":
+    parser = ArgumentParser(
+        "snapcraft_channel",
+        description="""
+    Determine the snapcraft channel based on the provided options and version.
+    """,
+    )
+
+    parser.add_argument(
+        "version", help="The version to determine the channel for.", type=Version.parse
+    )
+    parser.add_argument("--nightly", help="Use nightly channel", action="store_true")
+
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO)
+
+    version: Version = args.version
+    log.info(f"Parsed version: {version!r}")
+
+    tracks_for_version = get_tracks_for_version(version)
+    log.info(f"Tracks for version {version}: {', '.join(tracks_for_version)}")
+
+    if args.nightly:
+        print(
+            # Update the edge channel of the version's tracks
+            *(map(lambda track: f"--release={track}/{Channel.Edge}", tracks_for_version)),
+            # Update all nightly's channels
+            *(map(lambda chan: f"--release={Track.Nightly}/{chan}", CHANNELS)),
+            sep="\n",
+        )
+    else:
+        channel = get_channel_for_version(version)
+        log.info(f"Channel for version {version}: {channel}")
+
+        print(*(map(lambda track: f"--release={track}/{channel}", tracks_for_version)), sep="\n")
```

Comment:
Wouldn't be more useful to directly log the output of the script?
```suggestion
    tracks_for_version = get_tracks_for_version(version)

    release_channels = []
    if args.nightly:
        # Update channels "<tracks>/edge"
        release_channels.extend(map(lambda track: f"--release={track}/{Channel.Edge}", tracks_for_version))
        # Update channels "nightly/<risk>"
        release_channels.extend(map(lambda chan: f"--release={Track.Nightly}/{chan}", CHANNELS))
    else:
        channel = get_channel_for_version(version)
        release_channels.extend(map(lambda track: f"--release={track}/{channel}", tracks_for_version))

log.info(" ".join(release_channels))
print(*release_channels, sep="\n")
```

Reply:
done